Kernel/Process.cpp

```diff
@@ -626,6 +626,15 @@ void Process::unblock_waiters(Thread::WaitBlocker::UnblockFlags flags, u8 signal
 
 void Process::die()
 {
+    auto expected = State::Running;
+    if (!m_state.compare_exchange_strong(expected, State::Dying, AK::memory_order_acquire)) {
```

Comment:
Could this possibly cause a transition from  back Dead to Dying?

Reply:
No, the exchange only happens if `m_state == State::Running`, so if it's already at `State::Dead` this would fail.

Reply:
Ahh right because it's not a compare exchange weak loop, it won't keep reloading and rechecking expected. Carry on 🙂